### PR TITLE
Promote datavol over cloudinit

### PR DESCRIPTION
**Pull request by snecklifter**
##### SUMMARY

Volume boot order defaults to creation order in vm definition so we want to move datavolumes higher up than cloud-init volumes as its unlikely users will want to boot from these. Setting the boot order on a datavol is also non-trivial.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
kubevirt_vm

##### ADDITIONAL INFORMATION
Attempt to boot a kubevirt vm with both a datavol and a cloudinit vol. The cloudinit vol will be created first and therefore the vm attempts to boot from this and fails.

### CODE REVIEW

**Review by Akasurde — CHANGES_REQUESTED** [review]
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Comment by snecklifter** [comment]
> Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

Done.

**Comment by Andersson007** [comment]
cc @tima as a maintainer could you please take a look?

cc @machacekondra @mmazur @pkliczewski you were in the team_kubevirt in ansible/ansible before this stuff was moved to this repo. Are you interested in becoming maintainers of this repository?
There's the [maintainer guidelines](https://github.com/ansible/community-docs/blob/main/maintaining.rst) we've recently written and we can help if needed.

The failing check is unrelated, please ignore.

**Comment by Andersson007** [comment]
@pabelanger @Akasurde @gundalow  there's strange Zuul behavior. Do you know who we can ask for taking a look?
```
2021-09-01 07:47:50.431875 | TASK [Confirm collection can be imported into galaxy]
2021-09-01 07:47:54.098301 | centos-8 | Importing with galaxy-importer 0.4.0 Getting doc strings via ansible-doc ERROR: Error running ansible-doc: cmd="/usr/bin/env ANSIBLE_COLLECTIONS_PATHS=/tmp/tmpz4cdzb6s ANSIBLE_LOCAL_TEMP=~/.ansible/tmp ansible-doc --type module --json community.kubevirt.kubevirt_pvc community.kubevirt.kubevirt_cdi_upload community.kubevirt.kubevirt_template community.kubevirt.kubevirt_rs community.kubevirt.kubevirt_vm community.kubevirt.kubevirt_preset" returncode="1" b'[DEPRECATION WARNING]: Ansible will require Python 3.8 or newer on the \ncontroller starting with Ansible 2.12. Current version: 3.6.8 (default, Mar 19 \n2021, 05:13:41) [GCC 8.4.1 20200928 (Red Hat 8.4.1-1)]. This feature will be \nremoved from ansible-core in version 2.12. Deprecation warnings can be disabled\n by setting deprecation_warnings=False in ansible.cfg.\nERROR! module community.kubevirt.kubevirt_pvc missing documentation (or could not parse documentation): unknown doc_fragment(s) in file /tmp/tmpz4cdzb6s/ansible_collections/community/kubevirt/plugins/modules/kubevirt_pvc.py: community.kubernetes.k8s_auth_options\n' Finding content inside collection Loading module kubevirt_pvc Loading module kubevirt_cdi_upload Loading module kubevirt_template Loading module kubevirt_rs Loading module kubevirt_vm Loading module kubevirt_preset Loading module_utils kubevirt Loading doc_fragments kubevirt_vm_options Loading doc_fragments kubevirt_common_options Loading inventory kubevirt WARNING: Ignore files skip ansible-test sanity tests, found ignore-2.11.txt with 20 statement(s) WARNING: Ignore files skip ansible-test sanity tests, found ignore-2.9.txt with 5 statement(s) WARNING: Ignore files skip ansible-test sanity tests, found ignore-2.12.txt with 20 statement(s) WARNING: Ignore files skip ansible-test sanity tests, found ignore-2.10.txt with 20 statement(s) Collection loading complete Importer processing completed successfully
2021-09-01 07:47:54.740309 | centos-8 | ERROR
2021-09-01 07:47:54.740605 | centos-8 | {
2021-09-01 07:47:54.740721 | centos-8 |   "delta": "0:00:03.689799",
2021-09-01 07:47:54.740795 | centos-8 |   "end": "2021-09-01 07:47:54.102387",
2021-09-01 07:47:54.740893 | centos-8 |   "msg": "non-zero return code",
2021-09-01 07:47:54.740966 | centos-8 |   "rc": 1,
2021-09-01 07:47:54.741029 | centos-8 |   "start": "2021-09-01 07:47:50.412588"
2021-09-01 07:47:54.741090 | centos-8 | }
```

**Comment by Andersson007** [comment]
@pabelanger @Akasurde @gundalow maybe it relates to `ERROR! module community.kubevirt.kubevirt_pvc missing documentation (or could not parse documentation): unknown doc_fragment(s) in file /tmp/tmpz4cdzb6s/ansible_collections/community/kubevirt/plugins/modules/kubevirt_pvc.py`
I'll take a look

**Comment by Andersson007** [comment]
@snecklifter CI shoud be fixed, could you please rebase both your PRS and push?

**Comment by snecklifter** [comment]
> @snecklifter CI shoud be fixed, could you please rebase both your PRS and push?

Done!

**Comment by Andersson007** [comment]
@machacekondra @mmazur @pkliczewski could you please review this PR?

**Review by alinabuzachis — COMMENTED** [review]
@snecklifter Thank you for working on this.
I would suggest merging this PR https://github.com/ansible-collections/community.kubevirt/pull/19 first.

File: changelogs/fragments/promote-datavol-over-cloudinit.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - kubevirt_vm - Create datavol before cloudinit disk as its more likely to be a boot volume
```
Comment (alinabuzachis) [inline changelogs/fragments/promote-datavol-over-cloudinit.yaml]:
```suggestion
  - kubevirt_vm - Create datavol before cloudinit disk as its more likely to be a boot volume (https://github.com/ansible-collections/community.kubevirt/pull/20)
```

**Review by alinabuzachis — APPROVED** [review]

**Comment by Andersson007** [comment]
@Akasurde you requested changes, they are done. Can we merge this?

**Review by Akasurde — APPROVED** [review]
LGTM

**Comment by Andersson007** [comment]
@snecklifter thanks for the contribution! @alinabuzachis @Akasurde thanks for reviewing!:)

**Pull request merged**